vignettes/customizing-dccvalidator.Rmd

```diff
@@ -0,0 +1,69 @@
+---
+title: "Customizing dccvalidator"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{customizing-dccvalidator}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+dccvalidator is intended to be customizable for different settings. The Shiny
+application uses a [configuration file](https://github.com/Sage-Bionetworks/dccvalidator/blob/master/config.yml)
+to set details such as where to store uploaded files, which metadata templates
+to validate against, whom to contact with questions, etc.
+
+To create a custom version of the app, you'll need to follow these steps:
+
+1. Create a Synapse project or folder with the appropriate permissions to store
+   uploaded files. In AMP-AD, we created a folder to which consortium members
+   have permissions to read and write, but *not* download. Only the curation
+   team has the ability to download files (in order to assist with debugging).
+   See this
+   [example of how to create a project with appropriate permissions](https://github.com/Sage-Bionetworks/dccvalidator/blob/master/inst/app/create_project.R).
+1. Fork the [dccvalidator](https://github.com/Sage-Bionetworks/dccvalidator/blob/master/config.yml)
+   GitHub repository.
+1. Create a new configuration in the `config.yml` file. Note that any values you
+   do not customize will be inherited from the default configuration.
+1. (Optional): create a pull request with your configuration back to the
+   upstream dccvalidator repository.
+1. Within the file [`app.R`](https://github.com/Sage-Bionetworks/dccvalidator/blob/master/app.R),
+   replace the `"default"` configuration with the name of your new 
+   configuration.
+1. Deploy the application as described in the 
+   [Deploying dccvalidator](https://sage-bionetworks.github.io/dccvalidator/articles/deploying-dccvalidator.html) 
+   vignette.
+
+## Configuration options
+
+* `parent`: The Synapse project or folder where files will be stored
+* `study_table`: Synapse ID of a table that lists all of the existing studies in
+  the consortium
+* `annotations_table`: Synapse ID of a table that lists allowable annotation
```

Comment:
There is a required schema for this annotations table in order for the app not to complain. e.g. required keys, values for columnType. I suggest documenting that here.

Reply:
I believe this also applies to the study table.